content/blog/chess-knight-problem/index.md

```diff
@@ -0,0 +1,129 @@
+---
+{
+  title: "Chess Knight Problem: a quick and dirty solution in JavaScript",
+  description: "Here I present a quick and dirty solution to a common interview question where the solution is not nearly as complex as it may first appear.",
+  published: "2020-04-29T12:27:06.284Z",
+  authors: ["thodges314"],
+  tags: ["javascript", "computer science", "interview"],
+  attached: [],
+  license: "cc-by-4",
+}
+---
+
+This is a classic problem in Computer Science that is said to come up in various forms on programming interviews - given an 8 x 8 chessboard, a selected starting square, and a selected ending square, what is the minimum number of legal moves are required to get a knight piece from the starting square to the ending square? Note that this is _not_ the same as Knight's Tour, a much more involved problem where we try to find a path around the board such that the knight touches every square once without repeat.
+
+For some background - a chess knight moves in an L-shaped pattern - two up and one to the right, two to the left and one up, and so on. The knight jumps from it's starting location to it's ending location.
+
+There are many diagrams of possible knight moves on a google image search, but to avoid copyright claims, I made up my own diagram:
+![Image of possible moves for a knight centered on a chessboard.](./knight-moves-0.png)
+
+The red mark above is an arbitrary starting point, and the green marks are all of the possible places that the knight can jump from that point.
+
+# Solution Method {#solution-method}
+
+At first glance, this may look like a bizarre maze navigation algorithm with complex rules, and inspires any number of thoughts of number of possible iterations, how to decide if a move is constructive or not, etc.
+
+Happily, our solution is much simpler, such that I was able to tap one up in JavaScript in a single evening after I decided to look into this problem.
```

Comment:
I'm not sure I'd mention how you were able to get this done quickly. Might discourage those who're facing more difficulties in the progression and we don't want to discourage newcomers

Reply:
I'm leaving this in because the theme of the article, from the title, is that this solution is (and was) quick and dirty.  It used a small amount of code, but also didn't have any kind of elaborate web interface or anything.